src/extensions/mod.rs

```diff
@@ -86,11 +63,50 @@ impl<T: gtk::glib::IsA<gtk::Widget>> RelmWidgetExt for T {
     }
 }
 
+fn iter_children(widget: &gtk::Widget) -> impl Iterator<Item = gtk::Widget> {
+    let mut widget = widget.first_child();
+
+    std::iter::from_fn(move || {
+        if let Some(child) = widget.take() {
+            widget = child.next_sibling();
+            return Some(child);
+        }
+
+        None
+    })
+}
+
+/// Additional methods for `gtk::Box`.
+pub trait RelmBoxExt {
+    /// Returns all children of the box.
+    fn children(&self) -> Vec<gtk::Widget>;
+
+    /// Remove all children from the box.
+    fn remove_all(&self);
```

Comment:
`remove_all` appears more often and similar methods like `children`, too.

I don't want to put everything in one trait, yet where some widgets share the same functionality, a shared trait would make sense.

Maybe something like this:

```rust
trait RelmRemoveAll {
    fn remove_all(&self);
}

trait RelmChildren {
    type Child;
    fn children(&self) -> Vec<Child>;
}
```

Also, ideally every `children` method should return an iterator instead of an vector.

Reply:
We can have traits if we keep them as simple as possible.

Making `children` an iterator is a good idea.

Reply:
Take a look at the changes. I've added the traits as you have suggested. Also it's worth noting that `iter_children` method returns a `ChildrenIterator<Self>`, which itself implements `DoubleEndedIterator` and makes it possible to iterate from both sides of the container.